api/src/services/purged-docs.js

```diff
@@ -0,0 +1,161 @@
+const db = require('../db');
+const environment = require('../environment');
+const purgingUtils = require('@medic/purging-utils');
+const cache = require('./cache');
+const crypto = require('crypto');
+const logger = require('../logger');
+
+const purgeDbs = {};
+const getPurgeDb = (roles) => {
+  const hash = purgingUtils.getRoleHash(roles);
+  if (!purgeDbs[hash]) {
+    purgeDbs[hash] = db.get(purgingUtils.getPurgeDbName(environment.db, hash));
+  }
+  return purgeDbs[hash];
+};
+
+const getCacheKey = (roles, docIds) => {
+  const hash = crypto
+    .createHash('md5')
+    .update(JSON.stringify(docIds), 'utf8')
+    .digest('hex');
+
+  return `purged-${JSON.stringify(roles)}-${hash}`;
+};
+
+const regex = new RegExp(/^purged-.+-.+$/);
```

Comment:
This could use a more descriptive name. Also, I think you're passing a regex to the Regex constructor which is unnecessary?

Reply:
Removed as no longer needed. With the separate cache instances, we clear all keys when purge runs.